protocol/client/messenger.go

```diff
@@ -141,7 +141,8 @@ func (m *Messenger) Join(ctx context.Context, c Contact) error {
 	}
 
 	opts := protocol.DefaultRequestOptions()
-	if err := m.Request(ctx, c, opts); err != nil {
+	// NOTE(dshulyak) join ctx shouldn't have an impact on history timeout.
+	if err := m.Request(context.Background(), c, opts); err != nil {
```

Comment:
Maybe we should remove sending a request when joining and make it a separate action? It wouldn't be this PR change though.

Reply:
i think ideally it should be managed in background, so that we can do multiple attempts without blocking an API

the same when messenger is started